rule/rule.go

```diff
@@ -750,15 +750,20 @@ func (r *Rule) Args() []bzl.Expr {
 // Insert marks this statement for insertion at the end of the file. Multiple
 // statements will be inserted in the order Insert is called.
 func (r *Rule) Insert(f *File) {
-	// TODO(jayconrod): should rules always be inserted at the end? Should there
-	// be some sort order?
 	var stmt []bzl.Expr
 	if f.function == nil {
 		stmt = f.File.Stmt
+		r.index = len(stmt)
 	} else {
 		stmt = f.function.stmt.Body
+		var i int
+		for i = range f.Rules {
```

Comment:
`sort.Search` would be better than looping over the list.

This assumes the rules are sorted already though, right? I don't think they necessarily are.

Also what if the inserted rules aren't sorted? They're not required to be, in general.